torch/__init__.py

```diff
@@ -300,10 +300,69 @@ def set_default_dtype(d):
     _C._set_default_dtype(d)
 
 def set_deterministic(d):
-    r"""Sets a global flag to force all operations to use a deterministic
-    implementation if available. If an operation that does not have a
-    deterministic implementation is called while this setting is True, the
-    operation will throw a RuntimeError.
+    r"""Sets a global flag to force normally nondeterministic operations to
+    use a deterministic implementation, if available. Or if any of these
+    operations do not have a deterministic implementation, they will throw
+    a :class:`RuntimeError`.
+
+    .. warning::
+        This feature is a work in progress, so it does not affect every
+        nondeterministic operation yet. All affected operations are listed below.
+        Any other operations in PyTorch that call the listed operations
+        internally are also affected.
+
+    The following normally-nondeterministic operations will act
+    deterministically when `d=True`:
+
+        * Convolution operations
```

Comment:
I don't know if these groupings are helpful

Reply:
I've reduced the groupings to:
* Operations that will use a deterministic algorithm
* Operations that will throw an error
* Operations that will throw an error if `CUBLAS_WORKSPACE_CONFIG` is not set properly

I feel like these groupings are helpful, what do you think?

Reply:
Sounds really good